File: uaa/src/main/java/com/artaeum/uaa/config/ResourceServerConfig.java
package com.artaeum.uaa.config;

import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.http.SessionCreationPolicy;
import org.springframework.security.oauth2.config.annotation.web.configuration.EnableResourceServer;
import org.springframework.security.oauth2.config.annotation.web.configuration.ResourceServerConfigurerAdapter;

@Configuration
@EnableResourceServer
public class ResourceServerConfig extends ResourceServerConfigurerAdapter {

    @Override
    public void configure(HttpSecurity http) throws Exception {
        http
                .csrf()
                    .disable()
                .headers()
                    .frameOptions()
                    .disable()
                .and()
                    .sessionManagement()
                    .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
                .and()
                    .authorizeRequests()
                    .antMatchers("/register").permitAll()
                    .antMatchers("/activate").permitAll()
                    .antMatchers("/authenticate").permitAll()
                    .antMatchers("/account/reset-password/init").permitAll()
                    .antMatchers("/account/reset-password/finish").permitAll()
                    .antMatchers("/**").authenticated();
    }
}

File: storage/src/main/scala/com/artaeum/media/config/ResourceServerConfig.scala
package com.artaeum.media.config

import org.springframework.context.annotation.Configuration
import org.springframework.http.HttpMethod
import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.http.SessionCreationPolicy
import org.springframework.security.oauth2.config.annotation.web.configuration.{EnableResourceServer, ResourceServerConfigurerAdapter}

@Configuration
@EnableResourceServer
@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true)
class ResourceServerConfig extends ResourceServerConfigurerAdapter {

  override def configure(http: HttpSecurity): Unit = {
    http
      .csrf()
        .disable()
        .headers()
        .frameOptions()
        .disable()
      .and()
        .sessionManagement()
        .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
      .and()
        .authorizeRequests()
        .antMatchers(HttpMethod.GET).permitAll()
        .antMatchers("/**").authenticated()
  }
}
